packages/lib/gpc/src/gpcChecks.ts

```diff
@@ -0,0 +1,569 @@
+        const revealedValue = objClaims.entries[entryName];
+        if (revealedValue === undefined) {
+          throw new ReferenceError(
+            `Configuration reveals entry "${objName}.${entryName}" which` +
+              ` doesn't exist in claims.`
+          );
+        }
+      }
+
+      hasOwnerEntry ||= !!entryConfig.isOwnerCommitment;
+    }
+  }
+
+  // Reverse check that each revealed entry and object exists and is revealed
+  // in config.
+  for (const [objName, objClaims] of Object.entries(revealedClaims.pods)) {
+    const objConfig = boundConfig.pods[objName];
+    if (objConfig === undefined) {
+      throw new ReferenceError(
+        `Claims include object "${objName}" which doesn't exist in config.`
+      );
+    }
+    if (objClaims.entries !== undefined) {
+      for (const entryName of Object.keys(objClaims.entries)) {
+        const entryConfig = objConfig.entries[entryName];
+        if (entryConfig === undefined) {
+          throw new ReferenceError(
+            `Claims reveal entry "${objName}.${entryName}" which doesn't exist` +
+              ` in config.`
+          );
+        }
+
+        if (!entryConfig.isRevealed) {
+          throw new ReferenceError(
+            `Claims reveal entry "${objName}.${entryName}" which is not` +
+              ` revealed in config.`
+          );
+        }
+      }
+    }
+  }
+
+  // Check nullifier configuration.
+  if (revealedClaims.nullifierHash !== undefined && !hasOwnerEntry) {
+    throw new Error("Nullifier hash requires a defined owner entry.");
+  }
+}
+
+/**
+ * Checks that the required circuit parameters for a proof can be satisfied
+ * by a known circuit.  This is not an exact match, but instead each parameter
+ * of the chosen circuit must be greater than or equal to the required value.
+ *
+ * If the circuit name is not given, this will pick the check will pick the
+ * smallest known circuit which can satisfy the requirements.
+ *
+ * @param requiredParameters the minimum required parameter values
+ * @param circuitIdentifier a specific circuit to be used
+ * @returns the full description of the circuit to be used for the proof
+ * @throws Error if no known circuit can support the given parameters, or if
+ *   the named circuit cannot do so.
+ */
+export function checkCircuitParameters(
+  requiredParameters: ProtoPODGPCCircuitParams,
+  circuitIdentifier?: GPCIdentifier
+): ProtoPODGPCCircuitDesc {
+  if (circuitIdentifier !== undefined) {
+    const { familyName, circuitName } =
+      splitCircuitIdentifier(circuitIdentifier);
+    const foundDesc = ProtoPODGPC.findCircuit(familyName, circuitName);
+    if (foundDesc === undefined) {
+      throw new Error(`Unknown circuit name: "${circuitIdentifier}"`);
+    }
+    if (!ProtoPODGPC.circuitMeetsRequirements(foundDesc, requiredParameters)) {
+      throw new Error(
+        `Specified circuit "${circuitIdentifier}" does not meet proof requirements.`
```

Comment:
Do you have a suggestion?  None of the built-in JS error types I can find seem to fit this case, and I'm not sure it's worth defining a specific one for the purpose, given that I haven't seen many examples of catching errors by type rather than parsing strings.